lms/djangoapps/teams/views.py

```diff
@@ -570,3 +580,327 @@ def get(self, request, topic_id, course_id):
+
+            If the user is not staff and is not enrolled in the course
+            associated with the team they are trying to join, or if they are
+            trying to add a user other than themselves to a team, a 404 error
+            is returned. This is to prevent leaking information about the
+            existence of teams and users.
+
+            If the specified user does not exist, a 404 error is returned.
+
+            If the user is already a member of a team in the course associated
+            with the team they are trying to join, a 400 error is returned.
+            This applies to both staff and students.
+
+            If the user is not enrolled in the course associated with the team
+            they are trying to join, a 400 error is returned. This can occur
+            when a staff user posts a request adding another user to a team.
+    """
+
+    authentication_classes = (SessionAuthentication, OAuth2Authentication)
+    permission_classes = (permissions.IsAuthenticated,)
+
+    serializer_class = MembershipSerializer
+
+    paginate_by = 10
+    paginate_by_param = 'page_size'
+    pagination_serializer_class = PaginationSerializer
+
+    def get(self, request):
+        """GET /api/team/v0/team_membership"""
+        queryset = CourseTeamMembership.objects.all()
+
+        specified_username_or_team = False
+
+        if 'team_id' in request.QUERY_PARAMS:
+            specified_username_or_team = True
+            team_id = request.QUERY_PARAMS['team_id']
+            try:
+                team = CourseTeam.objects.get(team_id=team_id)
+            except CourseTeam.DoesNotExist:
+                return Response(status=status.HTTP_404_NOT_FOUND)
+            if not has_team_api_access(request.user, team.course_id):
+                return Response(status=status.HTTP_404_NOT_FOUND)
+            queryset = queryset.filter(team__team_id=team_id)
+
+        if 'username' in request.QUERY_PARAMS:
+            specified_username_or_team = True
+            if not request.user.is_staff:
+                enrolled_courses = (
+                    CourseEnrollment.enrollments_for_user(request.user).values_list('course_id', flat=True)
+                )
+                staff_courses = (
+                    CourseAccessRole.objects.filter(user=request.user, role='staff').values_list('course_id', flat=True)
+                )
+                valid_courses = [
+                    CourseKey.from_string(course_key_string)
+                    for course_list in [enrolled_courses, staff_courses]
+                    for course_key_string in course_list
+                ]
+                queryset = queryset.filter(team__course_id__in=valid_courses)
+            queryset = queryset.filter(user__username=request.QUERY_PARAMS['username'])
+
+        if not specified_username_or_team:
+            error_message = ugettext_noop('Username or team_id must be specified.')
```

Comment:
Nit: I know folks like ``ugettext_noop``, but with the extra variable and the pylint declaration, it seems harder to read to me. I don't see the trade off as being worth avoiding the duplicated string.

    return Response({
        'developer_message': 'Username of team_id must be specified',
        'user_message': gettext('Username of team_id must be specified')
    }, status=status.HTTP_400_BAD_REQUEST)